CMakeLists.txt

```diff
@@ -0,0 +1,250 @@
+if (NOT DEFINED CMAKE_EXPORT_COMPILE_COMMANDS)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+endif ()
+
+# Use folders to organise targets in IDEs via the FOLDER target
+# property.
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+
+#-----------------------------------------------------------------------
+# Project metadata
+
+string(CONCAT project_description
+    "An open-source interoperability standard for tools and content"
+    " management systems used in media production.")
+
+project(
+    OpenAssetIO
+    VERSION 0.0.0
+    DESCRIPTION ${project_description}
+    HOMEPAGE_URL https://github.com/TheFoundryVisionmongers/OpenAssetIO
+    LANGUAGES C CXX)
+
+
+#-----------------------------------------------------------------------
+# Default install directory
+
+# If the user hasn't configured cmake with an explicit
+# CMAKE_INSTALL_PREFIX, then set it to safely install into ./dist, to
+# help prevent the user from accidentally writing over /usr/local or
+# whatever.
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND PROJECT_IS_TOP_LEVEL)
+    set(CMAKE_INSTALL_PREFIX "${PROJECT_SOURCE_DIR}/dist" CACHE PATH "Installation location" FORCE)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Compiler convenience variables
+
+# This is a common check, so create a variable to make it more concise.
+# Note: this must come after `project(...)` otherwise
+# `CMAKE_CXX_COMPILER_ID` is undefined.
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(IS_GCC_OR_CLANG TRUE)
+else()
+    set(IS_GCC_OR_CLANG FALSE)
+endif()
+
+
+#-----------------------------------------------------------------------
+# Options
+
+# Installation options
+# TODO(DF): Install dir wrangling depending on desired Python/C++
+#   library combination. See OTIO.
+
+# Build options
+
+# Surface `BUILD_SHARED_LIBS` as a non-hidden CMake option in CMake
+# GUIs. This controls whether static or shared libs are built.
+# TODO(DF): We may end up needing to support building of both shared
+#   and static libs simultaneously. Also, using this means we inherit
+#   the option if openassetio is used as a CMake subproject.
+option(BUILD_SHARED_LIBS "Set to OFF to build static libraries" ON)
+
+# Default treating compiler warnings as errors to OFF, since
+# consumers of this project may use unpredictable toolchains.
+# For dev/CI we should remember to switch this ON, though!
+option(OPENASSETIO_WARNINGS_AS_ERRORS "Treat compiler warnings as errors" OFF)
+
+# Enable IPO by default. For such a small project, the increased build
+# resources seem worth the potential additional compiler optimisations.
+option(OPENASSETIO_ENABLE_IPO
+    "Enable Interprocedural Optimization, aka Link Time Optimization (LTO)" ON)
+
+# Enable position independent code by default. This is redundant for
+# shared libraries (CMake sets it automatically). Enabling this for
+# static libraries allows symbols to be exported by downstream
```

Comment:
Not so much exported, but linked into.

https://stackoverflow.com/questions/53086636/linking-a-static-library-into-a-shared-library-without-fpic

Reply:
Initially clarified in dad28440 but then I re-read the stackoverflow post, and I think that's saying that PIC is required for static libs to even be linked into a shared lib, even if none of its symbols are exported?  So assuming that's right, I tweaked the comment in 228f7b8d